src/app/(proper_react)/redesign/(authenticated)/user/dashboard/subscribed/page.tsx

```diff
@@ -48,13 +48,14 @@ export default async function Subscribed() {
   const scans = await getAllScanResults(profileId);
 
   // In dev mode, record scans every time this page is reloaded.
-  // The webhoook does this in production.
+  // The webhook does this in production.
   if (dev) {
     await addOnerepScanResults(
```

Comment:
@rhelmer I don't think this function (or even page) is in use anywhere, but it looks weird to me: we first fetch the scan results from the database using `getLatestOnerepScanResults`, and then insert those back into the database using `addOnerepScanResults`, if I understand it correctly.

Reply:
This page is linked to from the subscription flow directly right now (subplat is skipped, but it would be the callback page for subplat). However it's not final, it's something I put together a month ago so we could test if removal is working on stage/prod without having to go through subplat yet.

Doesn't this do `const scans = await getAllScanResults(profileId);` above (fetching the results from OneRep's API) and then add these to the database (because dev can't have webhooks)?

Reply:
Oh! OK, yeah, now I get it - so it does fetch the latest scan from the database, but only to obtain the scan ID, and then inserts the (newly fetched) scan results linked to that ID.

So I guess in that case, it should either:

- update the existing `onerep_scans` row to update its status, or
- add a new `onerep_scans` row that the new results would be linked to.

which I guess is another version of [this question](https://github.com/mozilla/blurts-server/pull/3436#discussion_r1334428132) for @mansaj.

Reply:
> Oh! OK, yeah, now I get it - so it does fetch the latest scan from the database, but only to obtain the scan ID, and then inserts the (newly fetched) scan results linked to that ID.

Reply:
I was pretty torn on this for the initial/existing implementation - in general I'd prefer to do things immutably (e.g. prefer `INSERT` over `UPDATE`), but `UPDATE` seemed natural for updating an existing scan.